### PR TITLE
feat(chat): add versioning for chat and prompt publications, fix freezes on admin UI and publish to path when unpublish entity from folders (Issues #321, #1966, #1991)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

add versioning for chat and prompt publications
fix freezes on admin UI
fix publish to path when unpublish entity from folders

Issues:

- #321
- #1966 
- #1991

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: success

**Review by irinakartun — DISMISSED** [review]

**Comment by nepalevov** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by YauheniyaH** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by YauheniyaH** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by YauheniyaH** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/utils/server/api.ts [hunk 1]
```diff
@@ -82,11 +115,31 @@ export const getPromptApiKey = (prompt: Omit<PromptInfo, 'id'>): string => {
...
-    name,
+  const parts = apiKey.split(pathKeySeparator);
+
+  const parsedApiKey: Omit<PromptInfo, 'folderId' | 'id'> = {
+    name: apiKey,
   };
+
+  if (options?.parseVersion) {
+    const version = parts.at(-1);
+    const validVersion = /^\d+\.\d+\.\d+(-\w+\.\d+)?/;
```
Comment (IlyaBondar) [inline apps/chat/src/utils/server/api.ts]: move out Regex out of method into constant

File: apps/chat/src/utils/server/api.ts [hunk 2]
```diff
@@ -67,12 +83,29 @@ export const parseConversationApiKey = (
...
+
+    if (version && validVersion.test(version)) {
+      parsedApiKey.publicationInfo = { version };
+      parsedApiKey.name = name
+        .split(pathKeySeparator)
+        .slice(0, -1)
+        .join(pathKeySeparator);
```
Comment (IlyaBondar) [inline apps/chat/src/utils/server/api.ts]: you can use `getPublicItemIdWithoutVersion` instead

File: apps/chat/src/utils/server/api.ts [hunk 3]
```diff
@@ -67,12 +83,29 @@ export const parseConversationApiKey = (
...
+  const parsedApiKey: Omit<ConversationInfo, 'folderId' | 'id'> = {
     model: { id: modelId },
     name,
     isPlayback: modelId === PseudoModel.Playback,
     isReplay: modelId === PseudoModel.Replay,
   };
+
+  if (options?.parseVersion) {
+    const version = parts.at(-1);
+    const validVersion = /^\d+\.\d+\.\d+(-\w+\.\d+)?/;
```
Comment (IlyaBondar) [inline apps/chat/src/utils/server/api.ts]: move out Regex out of method into constant

File: apps/chat/src/utils/server/api.ts [hunk 4]
```diff
@@ -45,11 +46,25 @@ const getModelApiIdFromConversation = (conversation: Conversation): string => {
...
+    conversation.publicationInfo.version !== NA_VERSION
+  ) {
+    return [
+      encodeModelId(
+        getModelApiIdFromConversation(conversation as Conversation),
+      ),
+      conversation.name,
+      conversation.publicationInfo.version,
+    ].join(pathKeySeparator);
+  }
```
Comment (IlyaBondar) [inline apps/chat/src/utils/server/api.ts]:
```suggestion
  const keyParts = [
    encodeModelId(getModelApiIdFromConversation(conversation as Conversation)),
    conversation.name,
  ];
  
  if (
    conversation.publicationInfo?.version &&
    conversation.publicationInfo.version !== NA_VERSION
  ) {
    keyParts.push(conversation.publicationInfo.version;
  }
  return keyParts.join(pathKeySeparator);
```

File: apps/chat/src/utils/server/api.ts [hunk 5]
```diff
@@ -82,11 +115,31 @@ export const getPromptApiKey = (prompt: Omit<PromptInfo, 'id'>): string => {
...
+  if (options?.parseVersion) {
+    const version = parts.at(-1);
+    const validVersion = /^\d+\.\d+\.\d+(-\w+\.\d+)?/;
+
+    if (version && validVersion.test(version)) {
+      parsedApiKey.publicationInfo = { version };
+      parsedApiKey.name = apiKey
+        .split(pathKeySeparator)
+        .slice(0, -1)
+        .join(pathKeySeparator);
```
Comment (IlyaBondar) [inline apps/chat/src/utils/server/api.ts]: you can use `getPublicItemIdWithoutVersion` instead

File: apps/chat/src/utils/server/api.ts [hunk 6]
```diff
@@ -67,12 +83,29 @@ export const parseConversationApiKey = (
...
-  return {
+  const parsedApiKey: Omit<ConversationInfo, 'folderId' | 'id'> = {
     model: { id: modelId },
     name,
     isPlayback: modelId === PseudoModel.Playback,
     isReplay: modelId === PseudoModel.Replay,
   };
+
+  if (options?.parseVersion) {
+    const version = parts.at(-1);
```
Comment (IlyaBondar) [inline apps/chat/src/utils/server/api.ts]:
```suggestion
    const version = parts.length > 2 && parts.at(-1);
```

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/utils/server/api.ts (hunk 6 above)
Comment (Alexander-Kezik) [reply apps/chat/src/utils/server/api.ts:94]: in this case NA_VERSION won't be handled

**Comment by YauheniyaH** [comment]
/deploy-review

>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by YauheniyaH** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1957.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Common/ShareIcon.tsx [hunk 7]
```diff
@@ -65,21 +69,22 @@ export default function ShareIcon({
...
           'absolute -bottom-1 -left-1 bg-layer-3',
           isPublished ? 'rounded-md' : 'rounded-sm',
+          iconWrapperClassName,
         )}
         data-qa={
           isPublished && isPublishingEnabled ? 'world-icon' : 'arrow-icon'
         }
       >
-        <Tooltip tooltip={t(isPublished ? 'Published' : 'Shared')}>
+        <Tooltip tooltip={t(isPublished ? t('Published') : t('Shared'))}>
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/ShareIcon.tsx]: it's already wrapped in translation
```suggestion
        <Tooltip tooltip={t(isPublished ? 'Published' : 'Shared')}>
```

**Review by IlyaBondar — APPROVED** [review]

**Review by irinakartun — APPROVED** [review]

**Pull request merged**